ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/deneb/helpers/MiscHelpersDeneb.java

```diff
@@ -60,29 +61,22 @@ private static KZG initKZG(final SpecConfigDeneb config) {
   /**
    * <a
    * href="https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/fork-choice.md#is_data_available">is_data_available</a>
+   *
+   * <p>In Deneb we don't need to retrieve data, everything is already available via blob sidecars.
    */
   @Override
-  public boolean isDataAvailable(
-      final UInt64 slot,
-      final Bytes32 beaconBlockRoot,
-      final List<KZGCommitment> kzgCommitments,
-      final List<BlobSidecar> blobSidecars) {
+  public boolean isDataAvailable(final List<BlobSidecar> blobSidecars) {
```

Comment:
What I don't like here, both methods, this and `validateBlobSidecarsAgainstBlock` could be called now with part of blobs, which is not clear. Next,  `isDataAvailable` is a spec thing, so I expect it will verify it completely. Plus the last piece of puzzle, that we call this method with blobSidecars with indexes matching all `kzgCommitments` from block is somewhere else in the 3rd place.    

I'm not sure what's the best solution for this. My suggestion (not the best one, need to think more on this) is:
1. rename this method, so it's actually 2nd round of verification
2. add comments to both methods that they could be used for partial verification
3. add actual isDataAvailable(kzgCommitmentsFromBlock, verifiedBlobKzgCommitments) here, which will finalize the action by just equal matching 2 lists

Reply:
or better isDataAvailable(block, verifiedBlobIdentifiers)
return IntRange.of(0, block.getKzgCommitments.size()).map(index -> verifiedBlobIdentifiers.contains(new BlobIdentifier(block.getRoot, index))).filter(false).findFirst().isEmpty()

Reply:
I agree that it isn't clear that those methods can be called with partial blobs. I'm thinking if it is better to go back to single method.

Consider that `ForkChoiceBlobSidecarsAvailabilityChecker::computeFinalValidationResult` is the method performing the checks on the final set.

Reply:
Yeah, yeah, I understand it's happened there but we'd better have it here it and by my understanding it should be called `isDataAvailable`